Backpack/Dialog/Classes/BPKDialogView.m

```diff
@@ -305,5 +300,20 @@ - (void)addButtonAction:(BPKDialogButtonAction *)action {
     [self setLayoutMargins];
 }
 
+#pragma mark - Private methods
+
+// This method helps us to set up the correct top anchor margin depending if we have a title or not
+- (void)updateTopAnchorConstraint {
+    [NSLayoutConstraint activateConstraints:@[
+        [[self hasTitle] ? self.titleLabel.topAnchor : self.descriptionLabel.topAnchor
```

Comment:
We tend to prefer using property syntax over method syntax
```suggestion
        [self.hasTitle ? self.titleLabel.topAnchor : self.descriptionLabel.topAnchor
```

Reply:
Will this not cause a bug if you unset and set the title multiple times since we never cleanup constraints? Also with this approach the `self.descriptionLabel.topAnchor` would have two competing constraints to other anchors? May I suggest that instead of this approach we keep the constraints as is, but add a height constriant that forces `titleLabel` to be `0pt` tall. With that we might also want to adjust the descriptionLabel top constraint with a negative constant

Reply:
I had different thoughts about this approach because I don't know why we have setters as I said in the PR comment in **Future Considerations**. 
Usually, those properties don't have to change once the Dialog is already presented I think or at least I don't remember that kind of requirements for a UI component in my experience, because if we expose those properties instead of injecting them through the designated initializer we need to react as you said to all the changes and run autolayout again and that also is not so cost-effective in terms of run time processing 

For now I can improve my current approach and experiment with your suggestion but I think that in the future a refactor would be better to avoid entirely this kind of situations.

Reply:
Cool, let's keep using this approach then. I'd like to make sure we aren't ending up with two competing constraints and ensure that we don't add hundreds of constraints if people keep setting and unsetting the title. 

I appreciate that this is unlikely to happen, but it feels like a subtle bug that we'd have to fight with down the line

Reply:
Also, the height is automatically set to 0 to the `titleLabel` if it doesn't have content or is nil so I think that if we change the descriptionLabel top constraint should be enough, I'll try it out